torch/csrc/jit/script/compiler.cpp

```diff
@@ -537,12 +537,12 @@ struct to_ir {
           << "range() expects one argument but got" << args.size();
     }
 
-    auto target_list = List<Ident>(targets);
+    auto target_list = List<Expr>(targets);
     if (target_list.size() != 1) {
       throw ErrorReport(stmt)
           << "Iteration variable unpacking is not supported";
     }
-    at::optional<Ident> target_ident{target_list[0]};
+    at::optional<Ident> target_ident{Var(target_list[0]).name()};
```

Comment:
This will throw a very ugly error message if `target_list[0]` is not a `Var`

Reply:
Right, I'll add a check that this is not a `Starred`. 

Btw apparently this is valid in python3 🤪:

```
for x, *y in (range(3), range(3), range(3)):
  print(x, y)